inference/src/translator/text_processor.cpp

```diff
@@ -106,6 +106,117 @@ void TextProcessor::process(std::string &&input, AnnotatedText &source, Segments
     }
   }
 }
+#elif defined(WASM)
+void TextProcessor::process(std::string&& input, AnnotatedText& source, Segments& segments) const {
+  /**
+   * This is an RAII guard for auto-freeing a malloc-allocated pointer when it leaves scope.
+   *
+   * The address of the internal pointer is passed to JavaScript where it is allocated
+   * via `malloc`, however we are responsible for deallocating the memory via `free` on
+   * the C++ side.
+   *
+   * Wrapping the deallocation in the checked destructor ensures that once the memory has been 
+   * allocated on the JS side, it is guaranteed to be deallocated when this stack frame is popped,
+   * whether from returning or from an exception being thrown.
+   */
+  struct ScopedPtr {
+    int32_t* ptr = nullptr;
```

Comment:
Question: Would this be more portable as a `uintptr_t`?

Reply:
I don't think that it would. The C++ reference [section on integer types](https://en.cppreference.com/w/cpp/types/integer) describes `uintptr_t` as an "unsigned integer type capable of holding a pointer to void."

My understanding is that `uintptr_t` is to be used when you need to treat the actual numerical value of an address as an integer to perform arithmetic operations on it. 

In this case, I truly am just creating a pointer to an array of signed 32-bit integers, and I don't particularly care about what the size of my addresses are (though, in Emscripten WASM32, they are also 32 bits). 

I need to assign these values from the JavaScript space, using `setValue()` with one of the listed [LLVM IR types](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html#access-memory-from-javascript), `i32` in this case. 

I think that this code is already portable in the sense that even if Emscripten were to hypothetically find a way to update all existing projects from WASM32 address space to a WASM64 address space, then I would still have an array of 32-bit integers here, even if my pointers to those integers have doubled in size. 

Since I'm not actually operating on the pointer's address value itself, I don't think that `uintptr_t` is the right type here. I truly want a pointer to `int32_t` regardless of the architecture.